### ISSUE TITLE
[Bug Report] Batched string-list input pads on the opposite side from HookedTransformer, and to_tokens(padding_side=...) is ignored

### ISSUE THREAD

**Issue by jlarson4** [TransformerLensOrg/TransformerLens#1690]
**Describe the bug**

Two related padding defects:

1. Opposite padding side on string lists – Passing a list of unequal-length strings to a compatibility-mode bridge produces logits that differ from `HookedTransformer` by 22.49 on gpt2 (logit scale ~126). Both models compute internally-correct activations, but at mirrored positions: HT right-pads, the bridge left-pads. Feeding pre-tokenized tensors instead gives an exact `0.0` match, which isolates the divergence to the string-list path. Every position index into the cache or logits for a padded row is therefore wrong after migration, `logits[0, -1]` is the last real token on the bridge and a pad token on HT.

2. `padding_side` is inert in `to_tokens` – `to_tokens(strs, padding_side="left")` and `padding_side="right"` return identical tensors; the tokenizer's own `padding_side` wins. The same argument on `HookedTransformer.to_tokens` genuinely changes the output. This also means `bridge.to_tokens(list)` cannot be used to reconstruct the indices that `bridge(list)` actually used.

**Code example**

```python
import torch
from transformer_lens import HookedTransformer
from transformer_lens.model_bridge import TransformerBridge

ht = HookedTransformer.from_pretrained("gpt2", device="cpu", dtype=torch.float32)
br = TransformerBridge.boot_transformers("gpt2", device="cpu", dtype=torch.float32)
br.enable_compatibility_mode()

batch = ["The quick brown fox", "Hello there, world! This is a longer sentence."]

print((ht(batch) - br(batch)).abs().max())                                  # 22.4894
print((ht(ht.to_tokens(batch)) - br(br.to_tokens(batch))).abs().max())      # 0.0

left  = br.to_tokens(batch, padding_side="left")
right = br.to_tokens(batch, padding_side="right")
print(torch.equal(left, right))                                             # True  <- ignored
print(torch.equal(ht.to_tokens(batch, padding_side="left"),
                  ht.to_tokens(batch, padding_side="right")))               # False <- honored
```

**System Info**

Installed from source on `dev`; macOS arm64, CPU, fp32, Python 3.12, transformers 5.13.0. Reproduced on gpt2, distilgpt2, and `roneneldan/TinyStories-1M`.

**Expected behaviour & fix pointer**

The two defects are coupled, details of my investigation below.

1. `to_tokens` must honor its own `padding_side` argument – It resolves the value
(`transformer_lens/model_bridge/transformer_bridge.py:760-761`, defaulting from the tokenizer)
and then never passes it to the `self.tokenizer(...)` call that follows, so the tokenizer's own
attribute decides. Threading the resolved value into that call fixes `to_tokens`,
`to_str_tokens` (`:812`), `get_token_position` (`:884`), and the forward paths (`:1455`,
`:1612`) in one place.

2. The batched-list forward should right-pad, matching `HookedTransformer`. – The forward
currently forces left-padding for any multi-item list (`:1735-1745`), which is where it diverges
from HT. Three pieces of evidence say right-padding is the correct default here:

- HT forces left-padding only in `generate()` (`transformer_lens/HookedTransformer.py:2007-2009`,
  "Force left-padding for batched generation"), not in its forward. Left-padding is a
  generation requirement, it puts the last real token at index `-1` for every row, and the
  bridge's `generate` should keep doing it. Extending it to `forward` is the deviation.
- The stated rationale does not hold for a causal forward. The code comment says left-padding
  is forced "so pad tokens don't contaminate attention or position embeddings", but with causal
  attention right-padding cannot contaminate anything: pads sit after the real tokens, so no
  real position can attend to them. Measured on gpt2 with a ragged batch, right-padded input
  reproduces the single-sequence logits for the real tokens exactly (0.0) both with and
  without an attention mask. Left-padding is what creates the need for mask and `position_ids`
  correction, because pads then precede the real tokens and shift every position.
- Left-padding is currently unavoidable and undiscoverable. Because the gate is
  `if _is_batched_list and padding_side is None`, setting `tokenizer.padding_side = "right"` is
  overridden (still left-pads), while passing `padding_side="right"` explicitly falls into the
  other branch, where defect 1 discards it. Neither route reaches right-padding, and
  `to_tokens` reports a layout the forward did not use.

Recommended shape of the fix: honor `padding_side` in `to_tokens`, drop the forced left-padding
from the batched-list forward so it follows the tokenizer/argument like every other path, and keep
the explicit left-padding in `generate`. Retain the auto-computed `attention_mask`/`position_ids` as they remain necessary whenever a caller does choose left-padding.

Acceptance:
- [ ] `to_tokens(padding_side="left"/"right")` produces different, correct tensors
- [ ] `bridge.to_tokens(list)` matches the token layout the string-list forward actually feeds the model
- [ ] Compatibility-mode logits for a ragged string list match `HookedTransformer` (or, if left-padding is kept, the deviation is documented in the migration guide's numerics section)
- [ ] `generate` still left-pads batched input (its correctness depends on it) — do not change that path
- [ ] Equal-length lists and single strings remain unaffected (currently exact)
- [ ] Setting `tokenizer.padding_side` is respected by the batched-list forward rather than overridden
- [ ] Test under `tests/unit/model_bridge/` covering ragged batches for both padding sides, asserting real-token activations match the single-sequence run
- [ ] `make unit-test` passes
- [ ] `uv run mypy .` passes

### Checklist
- [x] I have checked that there is no similar issue in the repo (**required**)
